uploadNeuroDB/bin/bestlinreg_claude.pl

```diff
@@ -0,0 +1,305 @@
+   ["-lsq9", "const", "-lsq9", \$opt{lsqtype},
+      "use 9-parameter transformation [default]" ],
+   ["-lsq12", "const", "-lsq12", \$opt{lsqtype},
+      "use 12-parameter transformation" ],
+   ["-mi", "const", "-mi", \$opt{objective},
+      "use mutual information as objective function [default -xcorr]" ],
+   ["-nmi", "const", "-nmi", \$opt{objective},
+      "use normalized mutual information as objective function [default -xcorr]" ]
+   );
+
+# Check arguments
+&Getopt::Tabular::SetHelp($Help, $Usage);
+&GetOptions (\@opt_table, \@ARGV) || exit 1;
+die $Usage if(! ($#ARGV == 2 || $#ARGV == 3));
+$source = shift(@ARGV);
+$target = shift(@ARGV);
+$outxfm = shift(@ARGV);
+$outfile = (defined($ARGV[0])) ? shift(@ARGV) : undef;
+
+# check for files
+die "$me: Couldn't find input file: $source\n\n" if (!-e $source);
+die "$me: Couldn't find input file: $target\n\n" if (!-e $target);
+if(-e $outxfm && !$opt{clobber}){
+   die "$me: $outxfm exists, -clobber to overwrite\n\n";
+   }
+if(defined($outfile) && -e $outfile && !$opt{clobber}){
+   die "$me: $outfile exists, -clobber to overwrite\n\n";
+   }
+
+my $mask_warning = 0;
+if( !defined($opt{source_mask}) ) {
+  $mask_warning = 1;
+} else {
+  if( !-e $opt{source_mask} ) {
+    $mask_warning = 1;
+  }
+}
+if( !defined($opt{target_mask}) ) {
+  $mask_warning = 1;
+} else {
+  if( !-e $opt{target_mask} ) {
+    $mask_warning = 1;
+  }
+}
+if( $mask_warning == 1 ) {
+  print "Warning: For optimal results, you should use masking.\n";
+  print "$Usage";
+}
+
+# make tmpdir
+$tmpdir = &tempdir( "$me-XXXXXXXX", TMPDIR => 1, CLEANUP => 1 );
+
+# set up filename base
+my($i, $s_base, $t_base, $tmp_xfm, $tmp_source, $tmp_target, $prev_xfm);
+$s_base = &basename($source);
+$s_base =~ s/\.mnc(.gz)?$//;
+$s_base = "S${s_base}";
+$t_base = &basename($target);
+$t_base =~ s/\.mnc(.gz)?$//;
+$t_base = "T${t_base}";
+
+# Mask the source and target once before blurring. Both masks must exist.
+
+my $source_masked = $source;
+my $target_masked = $target;
+
+if( defined($opt{source_mask}) and defined($opt{target_mask}) ) { 
+  if( -e $opt{source_mask} and -e $opt{target_mask} ) {
+    $source_masked = "${tmpdir}/${s_base}_masked.mnc";
+    &do_cmd( 'minccalc', '-clobber',
+             '-expression', 'if(A[1]>0.5){out=A[0];}else{out=A[1];}',
+             $source, $opt{source_mask}, $source_masked );
+
+    $target_masked = "${tmpdir}/${t_base}_masked.mnc";
+    &do_cmd( 'minccalc', '-clobber',
+             '-expression', 'if(A[1]>0.5){out=A[0];}else{out=A[1];}',
+             $target, $opt{target_mask}, $target_masked );
+  }
+}
+
+# initial transformation supplied by the user.
+
+$prev_xfm = ( defined $opt{init_xfm} && -e $opt{init_xfm} ) ?
+            $opt{init_xfm} : undef;
+
+# do a centering of the image first. Somehow, I think that minctracc should
+# be able to do this on its own.
+
+if( !defined( $prev_xfm ) ) {
+  my ($sxc,$syc,$szc) = split( ' ', `mincstats -quiet -com -world_only $source` );
```

Comment:
When running the backtick command, the shell arguments should be quoted (with `quotemeta`).